libraries/wallet/wallet.cpp

```diff
@@ -759,6 +759,76 @@ class wallet_api_impl
 
       return true;
    }
+
+   /**
+    * Get the required public keys to sign the transaction which had been
+    * owned by us
+    *
+    * NOTE, if `erase_existing_sigs` set to true, the original trasaction's
+    * signatures will be erased
+    *
+    * @param tx           The transaction to be signed
+    * @param erase_existing_sigs
+    *        The transaction could have been partially signed already,
+    *        if set to false, the corresponding public key of existing
+    *        signatures won't be returned.
+    *        If set to true, the existing signatures will be erase and
+    *        all required keys returned.
+   */
+   set<public_key_type> get_owned_required_keys( signed_transaction &tx,
+                                                    bool erase_existing_sigs = true)
+   {
+      set<public_key_type> pks = _remote_db->get_potential_signatures( tx );
+      flat_set<public_key_type> owned_keys;
+      owned_keys.reserve( pks.size() );
+      std::copy_if( pks.begin(), pks.end(),
+                    std::inserter( owned_keys, owned_keys.end() ),
+                    [this]( const public_key_type &pk ) {
+                       return _keys.find( pk ) != _keys.end();
+                    } );
+
+      if ( erase_existing_sigs )
+         tx.signatures.clear();
+
+      return _remote_db->get_required_signatures( tx, owned_keys );
+   }
+
+   signed_transaction sign_transaction2( signed_transaction tx,
+                                                 bool broadcast )
+   {
+      set<public_key_type> approving_key_set =
+         std::move( get_owned_required_keys( tx, false ) );
```

Comment:
I don't think the explicit `std::move` is needed here (as well as below). Although `=` is used here, this line is still an initialization, since the function returns a rvalue reference and `set` has a move constructor, the move constructor will be used by default.

Reply:
yeah, just a coding custom, I think this looks clearer